ormar/models/newbasemodel.py

```diff
@@ -171,12 +172,29 @@ def __setattr__(self, name: str, value: Any) -> None:  # noqa CCR001
         :return: None
         :rtype: None
         """
+        prev_hash = hash(self)
+
         if hasattr(self, name):
             object.__setattr__(self, name, value)
         else:
             # let pydantic handle errors for unknown fields
             super().__setattr__(name, value)
 
+        new_hash = hash(self)
+        if prev_hash == new_hash:
+            return
+
+        # Update all relation proxy caches with different hash
+        for relation in self._orm._relations.values():
```

Comment:
For review: There might be a better way to do this. Basically, I just wanted to make sure to update the cache when the current model changes--this code block basically makes sure it either updates its own cache, or updates the cache of its related model when it's in the related model list.